python/pyspark/sql/functions.py

```diff
@@ -2205,6 +2205,10 @@ def udf(f=None, returnType=StringType()):
         rows that do not satisfy the conditions, the suggested workaround is to incorporate the
         condition logic into the functions.
 
+    .. note:: Users can't rely on short-curcuit evaluation of boolean expressions to execute
+        conditionally user-defined functions too. For example, the two functions in an expression
+        like udf1(x) && udf2(y) will be both executed on all rows.
```

Comment:
does it apply to pandas_udf?

Reply:
I think pandas_udf doesn't use in boolean expressions as it returns pandas.Series.

Reply:
Sorry this is not correct. Pandas_udf can use in boolean expressions.